core/src/main/java/io/syndesis/core/KeyGenerator.java

```diff
@@ -32,40 +36,43 @@
  */
 public class KeyGenerator {
 
-    private static long lastTimestamp = System.currentTimeMillis();
+    private static AtomicLong lastTimestamp = new AtomicLong(currentTimeMillis());
     private static final byte randomnessByte;
-    private static long randomnessLong;
-
-    private KeyGenerator() {}
+    private static AtomicLong randomnessLong;
 
     static {
-        Random random = new Random();
+        final Random random = ThreadLocalRandom.current();
```

Comment:
Since this is an static block this method is only called once when the class is initialized. So still when multiple threads later call `createKey` they share the same Random object.

Wouldn't it be better to fetch the `Random` each time afresh in createKey() ?